R/adapter.r

```diff
@@ -232,17 +232,15 @@ construct_s3_adapter <- function() {
   write_function <- function(object, opts) {
     common_s3mpi_package_loader()
 
-    if (is.element('output', names(object))) {
```

Comment:
Just replace `names(object)` with `ls(object)` -- `ls` works for lists *and* environments, although it does not preserve order (which we don't care about here).